pkg/controller/controller.go

```diff
@@ -59,6 +59,12 @@ type Options struct {
 	// The overall is a token bucket and the per-item is exponential.
 	RateLimiter ratelimiter.RateLimiter
 
+	// NewWorkQueue constructs the queue for this controller once the controller is ready to start.
+	// This is a func because the standard Kubernetes work queues start themselves immediately, which
+	// leads to goroutine leaks if something calls controller.New repeatedly.
+	// Defaults to NewRateLimitingQueueWithConfig.
+	NewWorkQueue func(rateLimiter ratelimiter.RateLimiter) workqueue.RateLimitingInterface
```

Comment:
Maybe mention that the Ratelimiter is the Ratelimiter configured above?

Reply:
Oh and I am just realizing that we should probably pass the `name` as well?

Reply:
Makes sense, done!